parser.go

```diff
@@ -11,65 +11,60 @@ import (
 	"github.com/miekg/dns"
 )
 
-func matchingResult(qualifier tokenType) (SPFResult, error) {
-	if !qualifier.isQualifier() {
-		return SPFEnd, fmt.Errorf("invalid qualifier (%d)", qualifier)
-	}
-
-	var result SPFResult
-
+func matchingResult(qualifier tokenType) (Result, error) {
 	switch qualifier {
 	case qPlus:
-		result = Pass
+		return Pass, nil
 	case qMinus:
-		result = Fail
+		return Fail, nil
 	case qQuestionMark:
-		result = Neutral
+		return Neutral, nil
 	case qTilde:
-		result = Softfail
+		return Softfail, nil
+	default:
+		return internalError, fmt.Errorf("invalid qualifier (%d)", qualifier) // TODO it's fishy; lexer must reject it before
 	}
```

Comment:
nit: i think golinter was complaining about missing final return clause at the end of the function :(